protocols/gossipsub/src/protocol.rs

```diff
@@ -267,37 +267,25 @@ impl Encoder for GossipsubCodec {
     type Item = rpc_proto::Rpc;
     type Error = GossipsubHandlerError;
 
-    fn encode(&mut self, item: Self::Item, dst: &mut BytesMut) -> Result<(), Self::Error> {
-        let mut buf = Vec::with_capacity(item.encoded_len());
-
-        item.encode(&mut buf)
-            .expect("Buffer has sufficient capacity");
-
-        // length prefix the protobuf message, ensuring the max limit is not hit
-        self.length_codec
-            .encode(Bytes::from(buf), dst)
-            .map_err(|_| GossipsubHandlerError::MaxTransmissionSize)
+    fn encode(
+        &mut self,
+        item: Self::Item,
+        dst: &mut BytesMut,
+    ) -> Result<(), GossipsubHandlerError> {
+        Ok(self.codec.encode(item, dst)?)
     }
 }
 
 impl Decoder for GossipsubCodec {
     type Item = HandlerEvent;
     type Error = GossipsubHandlerError;
 
-    fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
-        let packet = match self.length_codec.decode(src).map_err(|e| {
-            if let std::io::ErrorKind::PermissionDenied = e.kind() {
-                GossipsubHandlerError::MaxTransmissionSize
-            } else {
-                GossipsubHandlerError::Io(e)
-            }
-        })? {
-            Some(p) => p,
-            None => return Ok(None),
+    fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>, GossipsubHandlerError> {
+        let rpc = match self.codec.decode(src).map_err(GossipsubHandlerError::Codec) {
+            Ok(Some(p)) => p,
+            _ => return Ok(None),
         };
```

Comment:
Should be able to use `?` here because you have a `From` implementation for the errors!


```suggestion
        let rpc = match self.codec.decode(src)? {
            Some(p) => p,
            None => return Ok(None),
        };
```

Reply:
Ahhh! Ok, that makes sense now. The `Ok()` that I had there was confusing me with the error message. Now that I've tried it this way I understand what the message was trying to tell me. Thanks!